### PR TITLE
[KOGITO-7267] - Consuming and producing events on Knative Eventing

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <zanini@redhat.com>

**JIRA:** https://issues.redhat.com/browse/KOGITO-7267

**Description:** In this PR, we introduce the new guide explaining the integration with Knative Eventing.

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributions doc](https://github.com/kiegroup/kogito-docs/blob/main/CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.9.x] KOGITO-XYZ Subject`
- [x] The nav.adoc file has a link to this guide in the proper category
- [ ] The index.adoc file has a card to this guide in the proper category, with a meaningful description

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -18,15 +18,14 @@
 *** xref:tooling/quarkus-dev-ui-extension/quarkus-dev-ui-workflow-instances-page.adoc[Workflow Instances in Kogito Serverless Workflow Tools extension]
 *** xref:tooling/quarkus-dev-ui-extension/quarkus-dev-ui-workflow-definition-page.adoc[Workflow Definitions in Kogito Serverless Workflow Tools extension]
 ** xref:tooling/kn-plugin-workflow-overview.adoc[Serverless Workflow plug-in for Knative CLI]
-// ** xref:tooling/tooling.adoc[Tooling]
```
Comment (fjtirado) [inline serverlessworkflow/modules/ROOT/nav.adoc:21]: is this needed?

Comment (ricardozanini) [reply serverlessworkflow/modules/ROOT/nav.adoc:21]: It's not. That's why I removed it.

**Review by fjtirado — APPROVED** [review]

File: serverlessworkflow/modules/ROOT/nav.adoc (hunk 1 above)
Comment (fjtirado) [reply serverlessworkflow/modules/ROOT/nav.adoc:21]: what is the color of the sky? ;)

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 2]
```diff
@@ -1 +1,290 @@
...
+:kubernetes_probes_url: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+:quarkus_k8s_url: https://quarkus.io/guides/deploying-to-kubernetes
+
+This document describes how you can configure Serverless Workflow to interact with link:{knative_eventing_doc_url}[Knative Eventing]. 
+
+Knative Eventing abstracts the events consumption via the link:{knative_eventing_doc_url}/sources[Source] and link:{knative_eventing_doc_url}/sinks[Sink] components. An event source is the Kubernetes object that produces the event. The sink is the Kubernetes object that receives the event. The workflow application can act as a source, a sink or both in the Knative Eventing platform.
+
+You need to add the Kogito Knative Eventing add-on dependency to indicate that you are using Knative Eventing. To enable Serverless Workflow to use Knative Eventing, add the following dependency to the `pom.xml` file of your project if using Maven:
+
+.Add dependency for Kogito Knative Eventing add-on in `pom.xml`
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:18]: maybe a tab on how adding this with maven, quarkus-cli and gradle?

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 3]
```diff
@@ -1 +1,290 @@
...
+[source,json]
+----
+<dependency>
+    <groupId>org.kie.kogito</groupId>
+    <artifactId>kogito-addons-quarkus-knative-eventing</artifactId>
+</dependency>
+----
+
+This add-on will take care of the required dependencies and any additional configurations that the workflow application needs to interact with the Knative Eventing platform.
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:27]:
```suggestion
This add-on will take care of the required dependencies and any additional configuration that the workflow application needs to interact with the Knative Eventing platform.
```

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 4]
```diff
@@ -1 +1,290 @@
...
+
+Although the default configuration that the Quarkus Knative Eventing add-on provides ought to be enough for the majority of uses cases, sometimes you may need to do additional configuration to serve a specific scenario.
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:29]:
```suggestion
Although, the default configuration that the Quarkus Knative Eventing add-on provides ought to be enough for the majority of uses cases, sometimes you may need to do additional configuration to serve a specific scenario.
```

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 5]
```diff
@@ -1 +1,290 @@
...
+
+[[con-knative-eventing-add-on-source-configuration]]
+== Knative Eventing add-on Source configuration
+
+The following properties are only valid if the workflow has at least one link:{spec_doc_url}#Event-Definition[event definition] of the type `produced`. That's when the workflow application produces events, acting as a Knative Source.
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:34]: these properties are the ones starting on line 44?
if so, ins't better to move this paragraph before the properties section?

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 6]
```diff
@@ -1 +1,290 @@
...
+
+=== HTTP Transport Configuration
+
+link:{knative_eventing_doc_url}/custom-event-source/sinkbinding/[Knative injects the `K_SINK` environment variable] in the workflow application when you deploy it in the cluster. Kogito uses this variable to address the produced events to the right Knative Sink.
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:38]:
```suggestion
link:{knative_eventing_doc_url}/custom-event-source/sinkbinding/[Knative Eventing] injects the `K_SINK` environment variable in the workflow application when you deploy it in the cluster. Kogito uses this variable to address the produced events to the right Knative Sink. 
```

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 7]
```diff
@@ -1 +1,290 @@
...
+
+|mp.messaging.outgoing.kogito_outgoing_stream.connector
+|`quarkus-http`
+|Quarkus Smallrye channel implementation used by Kogito. It's unlikely that you need to change this property.
+
+|===
+
+[NOTE]
+====
+Note that if the `K_SINK` variable is not present, the default value is `http://localhost:9000`. Feel free to override this default value in development environments.
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:56]:
```suggestion
If the `K_SINK` variable is not present, the default value is `http://localhost:9000`. Feel free to override this default value in development environments.
```

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 8]
```diff
@@ -1 +1,290 @@
...
+====
+
+=== Health Check Configuration
+
+By default, the application will generate a link:{kubernetes_probes_url}[health check probe] to verify if the Knative platform injected the `K_SINK` variable and it's valid. If the variable is not present, the pod won't be in ready state to receive requests.
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:61]:
```suggestion
By default, the application will generate a link:{kubernetes_probes_url}[health check probe] to verify if the Knative platform injected in the `K_SINK` variable is valid. If the variable is not present, the pod won't be in ready state to receive requests.
```

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 9]
```diff
@@ -1 +1,290 @@
...
+
+|org.kie.kogito.addons.knative.health_enabled
+|true
+|Whether to enable the health check to verify if the `K_SINK` variable has been injected in the environment.
+
+|===
+
+=== Knative Target Sink Generation Configuration
+
+Kogito Knative Eventing add-on generates a few Knative objects during build time. By default, it generates a link:{knative_eventing_doc_url}/broker[Knative Broker] named `default` if the workflow application is an Event Source.
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:75]:
```suggestion
The Kogito Knative Eventing add-on generates a few Knative objects during build time. By default, it generates a link:{knative_eventing_doc_url}/broker[Knative Broker] named `default` if the workflow application is an Event Source.
```

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 10]
```diff
@@ -1 +1,290 @@
...
+|true
+|Whether the add-on should generate a default Knative Broker in memory to sink and dispatch the messages. Turn this property to `false` in case you already have a broker installed in your namespace rather than the default one. Note that you can use `org.kie.kogito.addons.knative.eventing.sink.*` to configure your custom Sink. If not defined, this auto generated Broker will work as the Sink.
+
+|org.kie.kogito.addons.knative.sink.namespace
+|
+|Namespace where the generated Knative Sink is deployed.
+
+|org.kie.kogito.addons.knative.sink.api_version
+|eventing.knative.dev/v1
+|Namespace where the generated Knative Sink is deployed.
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:91]: is this description correct?

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 11]
```diff
@@ -1 +1,290 @@
...
+
+.Prerequisites
+
+Have a workflow application with the Knative Eventing Add-on.
+
+.Procedure
+
+1. Add the following link:{quarkus_k8s_url}[Quarkus Kubernetes extension] to your project:
++
+.Add dependency for Quarkus Kubernetes extension in `pom.xml`
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:138]: quarkus-cli and gradle options?

File: serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc [hunk 12]
```diff
@@ -1 +1,290 @@
...
+</dependency>
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-container-image-jib</artifactId>
+</dependency>
+----
++
+[TIP]
+====
+If you used the xref:tooling/kn-plugin-workflow-overview[kn workflow CLI] to create your project, this extension is already present.
```
Comment (spolti) [inline serverlessworkflow/modules/ROOT/pages/eventing/consume-produce-events-with-knative-eventing.adoc:153]:
```suggestion
If you have used the xref:tooling/kn-plugin-workflow-overview[kn workflow CLI] to create your project, this extension is already present.
```

**Review by spolti — COMMENTED** [review]
very nice!
just a few things, not mandatory though :)

**Review by ljmotta — COMMENTED** [review]
Very nice PR @ricardozanini ! I've made some comments inline.

**Review by spolti — APPROVED** [review]

**Comment by hmanwani-rh** [comment]
Merging this PR to create a patch PR with doc improvements.

**Pull request merged**